src/utils/proxy.ts

```diff
@@ -149,6 +149,16 @@ const getStatic = async function (pathname: string, publicFolder: string) {
   return `/${path.relative(publicFolder, file)}`
 }
 
+const isEndpointExists = async function (endpoint: string, origin: string) {
+  const url = new URL(endpoint, origin)
+  try {
+    const res = await fetch(url, { method: 'HEAD' })
```

Comment:
Do we check if the existing endpoint is external (https://...) or some existing dynamic route (localhost:3000/...).

Reply:
https://github.com/netlify/cli/pull/6994/files#diff-7c2533ca30d8693b3cecb3406646bc13b86b3969b43a0aa0659bb52c7676caa9R357

`options.target` matches the ssg itself e.g. 3000. If we'd do it to 8888, it will be pointless since we managing routes